### PR TITLE
Add .SLN and Project for VS 2015, Add wxWidgets 3.1.0 for VS 2015

### PR DESCRIPTION
**Pull request by Zangetsu38**
Add .SLN and Project for VIsual Studio 2015
Add wxWidgets 3.1.0 for Visual Studio 2015
Currently, six Project we need to be updated to be able compiled in Visual Studio 2015. 
This is "x86emitter, GSdx, xpad, utilities, pcsx2, SPU2-X"
Log error is here http://pastebin.com/t8MZ22DV
And I noticed that it was still the same error, except for GSdx: "error C2678: '==' binary: no operator found which takes a left operand of type 'const wxStringAllocator<wchar_t>' "

### CODE REVIEW

**Comment by Zangetsu38** [comment]
I need Help for fix last error

**Comment by avih** [comment]
This is a pretty bug patch.

I just tried to build using VS2015 (on a new system which didn't have VS2013 installed earlier, but I did install the DX june 2010 SDK), and PCSX2 and all plugins build just fine with two exceptions:
- in release (I tried AVX2) `padnull` failed to link with missing `libcpmt.lib`.
- In DEVEL* `padnull` and `dev9ghzdrk` both fail to link with missing `libcpmtd.lib`.

* DEVEL was also missing the soundtouch and opencl libs unless release is built first which builds these libs and puts them in a place which DEVEL builds can find them.

Bottom line, PCSX2 builds mostly just fine with the VS2013 sln file. So what's this PR really about? And how is wx3.1.0 related to this PR in any way?

**Comment by avih** [comment]
Not much of a talker? :)

**Comment by Zangetsu38** [comment]
If so, it's just that I have to redo the whole ^^

**Comment by avih** [comment]
Well, It was decided to stick to wx3.0.x since that the current release version. Of course, if wx3.0.x doesn't work with VS2015 then we'd need to rethink it.

It should be divided to few commits if you still want to do it:
1. One commit Just the VS2015 sln files.
2. Any updates, if required, to make all or most of the projects build (one commit per project).
3. Leave the wx update out of it for now, since wx2.8 builds fine with VS2015, and we'll handle the wx3.x on VS2015 once #657 is merged (move to wx3.x).

TBH, I'm not even sure we need VS2015 sln files (or if they're really different than VS2013 sln files). It worked for me out of the box, but maybe we do want them for some reason.

**Comment by Zangetsu38** [comment]
I'm on it.

**Comment by turtleli** [comment]
@avih - I get different errors?

The errors I get when I set VS2015 to use the v140_xp toolkit are:
 * GSdx - hash_map obsolete and to be removed, const element container errors
 * pthreads - timespec redefinition
 * xpad - hash_map obsolete and to be removed
 * wx2.8 errors

All are easily fixed (wx2.8 errors are solved by using wx3.0) and I had everything working last night. Lots of additional warnings from the build though, but most are about harmless things.

@Zangetsu38 - How is wx3.0.x incompatible with VS2015? Works fine here.

**Comment by Zangetsu38** [comment]
@turtleli Sorry but no, I Have error Syntax incorect with Wx 3.0

**Comment by turtleli** [comment]
And what errors are you getting? Because it works fine for me.

**Comment by Zangetsu38** [comment]
Syntax inccorect with wxconfig and vx++ need update for all, wx3.0 not compatible with vs 2015 that all

**Comment by avih** [comment]
Guys, can we take it one step at a time?

I _suggest_ to make this PR about VS2015 only with the current wx version (2.8).

We already have PR #657 for wx3.0 (at least for VS2013), and if required, we can have another PR for VS2015 fixes for wx3.0. Sounds good?

@turtleli  I didn't try to move to v140_xp. I just opened the vs2013 sln on a new system which has only VS2015 and DX SDK June 2010.

**Comment by Zangetsu38** [comment]
@avih not possible, wx 2.8 and 3.0 not works on VS 2015, obligatary use wx 3.1

**Comment by turtleli** [comment]
@Zangetsu38 - Those errors refer to wx2.8.
wx3.0 files are in 3rdparty/wxwidgets3.0/
wx2.8 files are in 3rdparty/wxWidgets, which is where your errors messages are coming from.

@avih - Ok, that most likely explains why I got errors and you didn't.

**Comment by Zangetsu38** [comment]
No, but you have not understood me, Wx 2.8 and 3.0 are fully compatible with Visual Studio 2015, are too old.
Only wx 3.1 which enables compiled on VS 2015, it's been a while that have known this.

**Comment by avih** [comment]
@Zangetsu38 , can you link to wx discussion or some other place which states that VS2015 is not compatible with anything below wx3.1? or do you base this assertion only on your own experiments?

**Comment by Zangetsu38** [comment]
@avih No discution, juste chat on irc

**Comment by gregory38** [comment]
https://forums.wxwidgets.org/viewtopic.php?t=40491

Seem doable. Vs2015 isn't final so maybe it will change again.

**Comment by Bigpet** [comment]
@gregory38 VS2015 got released the day before yesterday, so it is indeed final

**Comment by Zangetsu38** [comment]
@euuu ? sorry but yes, Visual studio 2015 is available on Final :)

**Comment by gregory38** [comment]
Oh I miss the news.

**Comment by Zangetsu38** [comment]
@gregory38 Microsoft Say Final version available 20 Juillet
